components/banners.js

```diff
@@ -0,0 +1,70 @@
+import Alert from 'react-bootstrap/Alert'
+import styles from './banners.module.css'
+import { useEffect, useState } from 'react'
+import { useMe } from '../components/me'
+import { useMutation } from '@apollo/client'
+import { WELCOME_BANNER_MUTATION } from '../fragments/users'
+import { useToast } from '../components/toast'
+
+export default function NewVisitorBanner () {
+  const me = useMe()
+  const toaster = useToast()
+  const [hidden, setHidden] = useState(true)
+  const handleClose = async () => {
+    window.localStorage.setItem('hideWelcomeBanner', true)
+    setHidden(true)
+    if (me) {
+      let error
+      try {
+        ({ error } = await hideWelcomeBanner())
+      } catch (e) {
+        error = e
+      }
+      if (error) toaster.danger(error.toString())
```

Comment:
This can be simplified like this:

```js
try {
  await hideWelcomeBanner()
} catch (err) {
  console.error(err)
  toaster.danger('mutation failed')
}
```

See `components/bookmark.js:28-34`